source/common/grpc/common.cc

```diff
@@ -289,11 +290,13 @@ std::string Common::typeUrl(const std::string& qualified_name) {
 }
 
 void Common::prependGrpcFrameHeader(Buffer::Instance& buffer) {
-  std::array<char, 5> header;
-  header[0] = 0; // flags
+
+  MemBlockBuilder<char> header(5);
+  header.appendOne(0); // flags
   const uint32_t nsize = htonl(buffer.length());
-  std::memcpy(&header[1], reinterpret_cast<const void*>(&nsize), sizeof(uint32_t));
-  buffer.prepend(absl::string_view(&header[0], 5));
+  header.appendData(
+      absl::Span<const char>(reinterpret_cast<const char*>(&nsize), sizeof(uint32_t)));
```

Comment:
let's revert this one back to its earlier state and use NOLINT(safe_memcpy). The previous impl does not do a dynamic memory allocation the new one does, so I'm hoping not to regress performance.

Also it looks like you have a memory leak below. You are calling releasePointer() but I don't see that pointer ever getting freed. This may account for your test problems. I'd recommend running tests locally first on this kind of change to save CI churn. This would be caught by `bazel test //test/...`